### DISCUSSION TITLE
How to recreate and create same look and feel of a GLB Model Viewer in Unity

### DISCUSSION THREAD

**Discussion by nikhilkraju** [google/model-viewer#4879] [Q&A]
In my project I have set up model loading functionality (using Trilib) into 3D from a JSON that can be augmented and placed in AR Similar to the GLB Editor . Assets in unity loads from S3 Buckets , I have set up exact Neutral Enviornment lighting and Reflections in unity , matched the material properties of GLB assets with the assets loading in unity (Metallic values , roughness and smoothness ) . 
- I have tried Standard pipeline as well as URP Pipeline with minimal post-processing Effects to match the Look and feel of the GLB editor . 
- GLB models cannot be loaded directly due to package conflicts between URP and gltf , as most of the assets have draco compression the glb assets cannot be extracted properly .
- Currently the model viewer project is setup to load FBX Assets as assetbundles . 

Problem : 
- Textures are not matching as per the GLB Viewer though exact material properties from the asset in glb editor is applied in unity 
- Colour variations are there on every asset (total 50 + assets are there)
- If we tweak the post processing settings to make one asset match with the GLB Viewer , other assets are affected 
- Universal lighting setting / standardised lighting setting in unity for all the assets to match the Look and feel of those assets in GLB Viewer is being the hurdle .

Is there anyway to achieve the exact look and feel of the GLB Editor models inside unity via tweaking and setting up the project setting 
 or any other way , please help !
 
 @jasondavies @gravityrail @bsdorra @mrdoob @pushmatrix

**Comment by elalish** [comment]
You'd probably be better off asking on the Unity forums - also, some images would help a lot in conveying what actual differences you're seeing. There are tools like `gltf-pipeline` on npm that can remove DRACO encoding, if that's a problem. There are also multiple plugins for glTF in Unity - which ones have you tried?

**Reply by nikhilkraju** [reply]
We tried that @elalish , It Works for some models but not for all . We stopped using GLB Files and rather went with FBX .

I am using URP sample demo project with high configurations.
I have added the same HDRI setup in my Unity scene.

To match the rendering view with google model viewer, I am applying post-processing, but with it actual color of the model is changing. even when I am using Unlit textures then also due to post processing it’s changing.

global volume

in Color adjustments: contrast, saturation, post exposure.

in Color Curves: Master curve, red hue curve.

also tried comparing both modes tone-mapping mode, and both has slight differences in color mainly.Unity Viewer Neutral VS Godel Viewer Neutral Unity Viewer ACES VS Google viewer ACES

Lighting settings

Environmental lights, Neutral HDRI and reflections with realtime GI.

HDRI’s exposure and angle mapped exactly. color difference exist there also.

Maybe I should not use post processing, but then model looks very off-color/grey/bright etc.
And with Post-processing mainly color is getting messy.

Should I use some other configuration rather then post-processing?

Attaching few screenshots of the lighting setup and the postprocessing in the project for more insight :

https://docs.google.com/document/d/1_A0C2hyN9Ta4K1LUSBB3LIVhurjGOi0cR6MCcwIYRCg/edit?usp=sharing

**Reply by nikhilkraju** [reply]
@pushmatrix @gravityrail @jasondavies

**Reply by elalish** [reply]
FBX will tend to be harder to match than GLB, since it's less standardized. I don't know exactly what kind of post-processing you're doing, but I'd recommend looking closely at tone mapping settings, as that makes probably the biggest difference to color: https://modelviewer.dev/examples/lightingandenv/#toneMapping.
